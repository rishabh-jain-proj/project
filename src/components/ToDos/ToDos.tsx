import React from 'react';

interface List {
    id: number,
    todo: string,
    status: boolean
}

interface Props {
    todos: List[];
    softDelete: (id: number) => void;
}

export const ToDos: React.FC<Props> = ({ todos, softDelete }) => {
    const todoList = todos.length ? (
        todos.map(todo => {
            return (
                <div style={{ textDecoration:  !todo.status ? 'line-through' : '', cursor: 'pointer' }}
                    onClick={() => softDelete(todo.id)}
                    className="collection-item"
                    key={todo.id}
                    title={todo.status ? 'Click To Mark As Completed': 'Click To UnMark As Incomplete'}>
                    <li className="todo-list">{todo.todo}</li>
                </div>
            )
        })) : (<p data-testid="empty-msg" className="center">Start Adding Your Todos...</p>)
    return (
        <div className="todos">
            {todoList}
        </div>
    )
}