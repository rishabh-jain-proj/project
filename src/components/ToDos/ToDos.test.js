import React from 'react';
import ReactDom from 'react-dom';
import Enzyme, { mount } from 'enzyme';
import EnzymeAdapter from "enzyme-adapter-react-16";
import { ToDos } from './ToDos';
import { render, cleanup } from '@testing-library/react';
import "@testing-library/jest-dom/extend-expect";


Enzyme.configure({
    adapter: new EnzymeAdapter(),
    disableLifecycleMethods: true
});

describe('ToDos component testing', () => {
    afterEach(cleanup);
    let wrapper;
    let todos = [{ id: 1, todo: 'Rishabh Jain - Adding Test', status: true }]

    it("should render without crashing", () => {
        const div = document.createElement("div");
        ReactDom.render(<ToDos todos={[]} softDelete={jest.fn()} />, div);
        ReactDom.unmountComponentAtNode(div);
    });

    it("should render ToDos correctly, with empty Todos Data", () => {
        wrapper = render(<ToDos todos={[]} softDelete={jest.fn()} />)
        expect(wrapper).toMatchSnapshot();
        const { getByTestId } = wrapper;
        expect(getByTestId('empty-msg'))
            .toHaveTextContent("Start Adding Your Todos...")
    });


    it('should Render ToDos List', () => {
        let softDelete = jest.fn();
        wrapper = mount(<ToDos todos={todos} softDelete={jest.fn()} />);
        let allToDoListELe = wrapper.find('.todo-list');
        expect(allToDoListELe).toHaveLength(todos.length)
        expect(allToDoListELe.hostNodes().render().text())
            .toEqual(todos[0].todo);
    }); 
});