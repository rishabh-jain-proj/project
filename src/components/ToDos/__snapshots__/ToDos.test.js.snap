// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ToDos component testing should render ToDos correctly, with empty Todos Data 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <div
        class="todos"
      >
        <p
          class="center"
          data-testid="empty-msg"
        >
          Start Adding Your Todos...
        </p>
      </div>
    </div>
  </body>,
  "container": <div>
    <div
      class="todos"
    >
      <p
        class="center"
        data-testid="empty-msg"
      >
        Start Adding Your Todos...
      </p>
    </div>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
